uspace/lib/c/generic/uuid.c

```diff
@@ -64,8 +64,8 @@ errno_t uuid_generate(uuid_t *uuid)
 	}
 
 	/* Version 4 UUID from random or pseudo-random numbers */
-	uuid->b[8] = (uuid->b[8] & ~0xc0) | 0x40;
-	uuid->b[6] = (uuid->b[6] & 0xf0) | 0x40;
+	uuid->b[6] = (uuid->b[6] & 0x4F) | 0x40;
+	uuid->b[8] = (uuid->b[8] & 0xBF) | 0xB0;
```

Comment:
This is incorrect. I committed a proper fix to the mainline.

Reply:
To elaborate, `| 0xB0` is setting more bits than it should. The rest is ok, but my change also has slightly different `&` masks as a matter of personal preference.

Reply:
You're right, the `| 0xB0` part would end up making it exactly B - which of course is wrong since it can be 8, 9, A or B. It supposed to be a fallback when `& 0xB` ends up being `0x0`. It should be therefore `| 0x80`